lib/chef/resource/hostname.rb

```diff
@@ -231,27 +231,25 @@ def updated_ec2_config_xml
             if xml_contents.empty?
               Chef::Log.warn('Unable to properly parse and update C:\Program Files\Amazon\Ec2ConfigService\Settings\config.xml contents. Skipping file update.')
             else
-              declare_resource(:file, 'C:\Program Files\Amazon\Ec2ConfigService\Settings\config.xml') do
+              file 'C:\Program Files\Amazon\Ec2ConfigService\Settings\config.xml' do
```

Comment:
Not really your responsibility but it feels like this string should be a constant...

Reply:
let me hit it with chefstyle

Reply:
actually that's to avoid having to escape the \ so that's intentional

Reply:
I mostly meant to avoid repeating the file name in this file.

Reply:
Updated